src/network/serverpackethandler.cpp

```diff
@@ -1277,21 +1279,19 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
 				}
 			}
 
-			/* Actually dig node */
+			if (!is_valid_dig || n.getContent() == CONTENT_IGNORE)
+				return;
 
-			if (is_valid_dig && n.getContent() != CONTENT_IGNORE)
-				m_script->node_on_dig(p_under, n, playersao);
+			// Actually dig node
+			m_script->node_on_dig(p_under, n, playersao);
 
-			v3s16 blockpos = getNodeBlockPos(floatToInt(pointed_pos_under, BS));
-			RemoteClient *client = getClient(pkt->getPeerId());
-			// Send unusual result (that is, node not being removed)
-			if (m_env->getMap().getNodeNoEx(p_under).getContent() != CONTENT_AIR) {
-				// Re-send block to revert change on client-side
-				client->SetBlockNotSent(blockpos);
-			}
-			else {
-				client->ResendBlockIfOnWire(blockpos);
-			}
+			/*
+				If the node is still the same, e.g. because of protection,
+				send it to the client to undo dig prediction
+			*/
+			MapNode newnode = m_env->getMap().getNodeNoEx(p_under);
+			if (newnode.getContent() == n.getContent())
```

Comment:
doesn't work with e.g. `node_dig_prediction = "the:very_same_node"`,
the server needs to apply the "full" dig prediction here and only send if the result differs

Reply:
I don't understand you. When the node differs, nothing needs to be sent because a TOCLIENT_REMOVENODE or something similar is already sent because of digging.
If the node is not changed, the dig prediction of the client may needs to be undone.
If dig prediction is disabled (or set to the same node), sending the current node is not required, but I think explicitly disabling is not needed because it's an edge case.

Reply:
well if you think this edge case is not important then so be it